components/tidb_query_expr/src/impl_json.rs

```diff
@@ -226,6 +227,16 @@ fn valid_paths(expr: &tipb::Expr) -> Result<()> {
     Ok(())
 }
 
+fn unquote_string(s: &str) -> Result<String> {
```

Comment:
Why do we add a dedicated function to do this? Can’t we reuse the original logic?

Reply:
The original `unquote_json` doesn't remove first and last double quote, which is not compatible with MySQL.

    mysql> select json_unquote('"abc"');
    +-----------------------+
    | json_unquote('"abc"') |
    +-----------------------+
    | abc                   |
    +-----------------------+

So I add this one to remove.